### PR TITLE
Add Managed Kafka Cluster resource and tests

### PR DESCRIPTION
**Pull request by jessdejong**
Description: Adding a new Managed Kafka Cluster terraform resource to terraform-provider-google-beta. Includes a basic create, basic update, and CMEK test.

Issue: https://b.corp.google.com/issues/322841720

```release-note:new-resource
`google_managed_kafka_cluster` (beta)
```

### CODE REVIEW

**Comment by jessdejong** [comment]
Note: Sam Levenick from the Google terraform team has said they will review this PR.

Also, [this PR](https://github.com/GoogleCloudPlatform/magic-modules/pull/10783) also ran into the same issue where the # of affected tests > 500. Fixed when /gcbrun was run.

**Review by slevenick — CHANGES_REQUESTED** [review]
CMEK test needs a role granted to the service account that uses the key

File: mmv1/products/managedkafka/Cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                - !ruby/object:Api::Type::String
+                  name: subnet
+                  description: "Name of the VPC subnet from which the cluster is accessible. Both broker and
+                    bootstrap server IP addresses and DNS entries are automatically created
+                    in the subnet. The subnet must be located in the same region as the
+                    cluster. The project may differ. A minimum of 1 subnet is required.
+                    A maximum of 10 subnets can be specified. The name of the subnet must be
+                    in the format `projects/PROJECT_ID/regions/REGION/subnetworks/SUBNET`."
+                  required: true
+                  diff_suppress_func: 'tpgresource.ProjectNumberDiffSuppress'
```
Comment (slevenick) [inline mmv1/products/managedkafka/Cluster.yaml:54]: Do we have an ETA on this being fixed server-side?

File: mmv1/templates/terraform/examples/managedkafka_cluster_cmek.tf.erb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  provider = google-beta
+}
+
+resource "time_sleep" "wait_50_seconds" {
+  depends_on = [google_kms_crypto_key_iam_binding.crypto_key_binding]
+  create_duration = "50s"
+}
+
+#resource "google_service_account" "gmk_service_account" {
```
Comment (slevenick) [inline mmv1/templates/terraform/examples/managedkafka_cluster_cmek.tf.erb:27]: Remove this if it's not being used

File: mmv1/templates/terraform/examples/managedkafka_cluster_cmek.tf.erb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  provider = google-beta
+}
+
+resource "google_kms_crypto_key_iam_binding" "crypto_key_binding" {
+  crypto_key_id = google_kms_crypto_key.key.id
+  role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+
+  members = [
+    # "serviceAccount:service-${data.google_project.project.number}@gcp-sa-staging-managedkafka.iam.gserviceaccount.com",
```
Comment (slevenick) [inline mmv1/templates/terraform/examples/managedkafka_cluster_cmek.tf.erb:63]: Remove if unused

**Review by slevenick — COMMENTED** [review]
Couple minor questions, overall looks fine

**Comment by slevenick** [comment]
/gcbrun hopefully some more of these tests will be fixed. They're unrelated so I'll merge without them all passing

**Review by slevenick — APPROVED** [review]
LGTM!

**Pull request merged**